src/libraries/System.Net.Http.WinHttpHandler/tests/FunctionalTests/WinHttpHandlerTest.cs

```diff
@@ -69,6 +71,79 @@ public async Task GetAsync_ConcurrentRead_ThrowsInvalidOperationException()
+            await RemoteExecutor.Invoke(async (version, uri) =>
+            {
+                AppContext.SetSwitch("System.Net.Http.UseWinHttpCertificateCaching", true);
+                int callbackCount = 0;
+                var handler = new WinHttpHandler()
+                {
+                    ServerCertificateValidationCallback = (_, _, _, _) =>
+                    {
+                        Interlocked.Increment(ref callbackCount);
+                        return true;
+                    }
+                };
+                using (var client = new HttpClient(handler))
+                {
+                    for (int i = 0; i < 5; i++)
+                    {
+                        var response = await client.SendAsync(new HttpRequestMessage(HttpMethod.Get, uri)
+                        {
+                            Version = Version.Parse(version)
+                        });
+                        Assert.Equal(HttpStatusCode.OK, response.StatusCode);
+                        _ = await response.Content.ReadAsStringAsync();
+                    }
+                    Assert.Equal(1, callbackCount);
+                }
+            }, version.ToString(), uri.ToString()).DisposeAsync();
+        }
+
+        [OuterLoop]
+        [ConditionalTheory(typeof(RemoteExecutor), nameof(RemoteExecutor.IsSupported))]
+        [MemberData(nameof(HttpVersions))]
+        public async Task SendAsync_ServerCertificateValidationCallbackCertificateTimerTriggered_CalledTwice(Version version, Uri uri)
+        {
+            await RemoteExecutor.Invoke(async (version, uri) =>
+            {
+                const int certificateCacheCleanupInterval = 10;
+                AppContext.SetSwitch("System.Net.Http.UseWinHttpCertificateCaching", true);
+                AppDomain.CurrentDomain.SetData("System.Net.Http.WinHttpCertificateCachingCleanupTimerInterval", certificateCacheCleanupInterval);
+                AppDomain.CurrentDomain.SetData("System.Net.Http.WinHttpCertificateCachingStaleTimeout", certificateCacheCleanupInterval);
+                int callbackCount = 0;
+                var handler = new WinHttpHandler()
+                {
+                    ServerCertificateValidationCallback = (_, _, _, _) =>
+                    {
+                        Interlocked.Increment(ref callbackCount);
+                        return true;
+                    }
+                };
+                using (var client = new HttpClient(handler))
+                {
+                    var response = await client.SendAsync(new HttpRequestMessage(HttpMethod.Get, Configuration.Http.SecureRemoteEchoServer)
+                    {
+                        Version = Version.Parse(version)
+                    });
+                    Assert.Equal(HttpStatusCode.OK, response.StatusCode);
+                    _ = await response.Content.ReadAsStringAsync();
+                    await Task.Delay(TimeSpan.FromMilliseconds(certificateCacheCleanupInterval * 3));
```

Comment:
As written, this is very likely to be very flaky in CI. A timer could easily be delayed by a few dozen ms and make this fail.

Consider writing it along the lines of
```c#
SetTimeout(10);

for (int i = 0; i < 5 && callbackCount <= 1; i++)
{
    await Task.Delay(100 * i);
    await DoRequest();
}

Assert.True(callbackCount > 1);
```